.github/workflows/compile-requirements.yml

```diff
@@ -0,0 +1,82 @@
+name: Generate requirement files
+
+on:
+  workflow_call:
+    inputs:
+      os:
+        default: '["ubuntu-latest"]'
+        description: A string representation of a list of operating systems.
+        required: false
+        type: string
+      python:
+        default: '["3.10"]'
+        description: A string representation of a list of Python versions.
+        required: false
+        type: string
+      requirements_directory:
+        default: requirements
+        description: The directory where requirements.in is located.
+        required: false
+        type: string
+      requirements_files:
+        default: '["requirements.in"]'
+        description: A string representation of a list of requirement files.
+        required: false
+        type: string
+
+jobs:
+  compile-requirements:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: ${{ fromJSON(inputs.os) }}
+        python-version: ${{ fromJSON(inputs.python) }}
+        requirements_file: ${{ fromJSON(inputs.requirements_files) }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install --upgrade pip-tools
+      - name: Generate
+        # Generate filename based on Python version and OS.
+        run: |
+          cd ${{ inputs.requirements_directory }}
```

Comment:
why is this not `fromJSON(inputs.requirements_directory)`?

Reply:
`inputs.requirements_directory` is a string, and doesn't need to be converted back to a list or anything else.